components/ai_chat/core/browser/model_service.cc

```diff
@@ -259,6 +263,26 @@ void ModelService::MigrateProfilePrefs(PrefService* profile_prefs) {
         profile_prefs->SetString(kDefaultModelKey, kDefaultModelBasicTo);
       }
     }
+
+    // Ensure that custom models have a model_system_prompt key
+    base::Value::List custom_models_list =
+        profile_prefs->GetList(kCustomModelsList).Clone();
+
+    // Flag to check if any updates were made
+    bool models_updated = false;
+
+    for (auto& model_value : custom_models_list) {
+      if (base::Value::Dict* model_dict = model_value.GetIfDict()) {
+        if (!model_dict->FindString(kCustomModelSystemPromptKey)) {
+          model_dict->Set(kCustomModelSystemPromptKey, "");
+          models_updated = true;
+        }
+      }
+    }
```

Comment:
optional: maybe it makes sense to allow null (or `std::nullopt`) here, as conceptually the empty string and `null` feel different (one is that there is no custom prompt, and the other is that the custom prompt is empty)

Also, we wouldn't need this migration code